providers/flagd/pom.xml

```diff
@@ -86,13 +92,14 @@
               <goal>exec</goal>
             </goals>
 						<configuration>
-							<!-- run: git submodule update \-\-init \-\-recursive -->
+							<!-- run: git submodule update \-\-init \-\-recursive \-\-remote -->
 							<executable>git</executable>
 							<arguments>
 								<argument>submodule</argument>
 								<argument>update</argument>
 								<argument>--init</argument>
 								<argument>--recursive</argument>
+								<argument>--remote</argument>
```

Comment:
```suggestion
```

Adding this param make the build non-deterministic. We should only pull the remote intentionally, when we want to update the submodule.